hbase-it/src/test/java/org/apache/hadoop/hbase/HBaseClusterManager.java

```diff
@@ -159,6 +169,38 @@ public void execute() throws IOException {
     }
   }
 
+  /**
+   * Executes commands over SSH
+   */
+  protected class RemoteSudoShell extends Shell.ShellCommandExecutor {
+    private String hostname;
+
+    public RemoteSudoShell(String hostname, String[] execString, long timeout) {
+      this(hostname, execString, null, null, timeout);
+    }
+
+    public RemoteSudoShell(String hostname, String[] execString, File dir, Map<String, String> env,
+        long timeout) {
+      super(execString, dir, env, timeout);
+      this.hostname = hostname;
+    }
+
+    @Override
+    public String[] getExecString() {
+      String at = sshUserName.isEmpty() ? "" : "@";
+      String remoteCmd = StringUtils.join(super.getExecString(), " ");
+      String cmd = String.format(tunnelSudoCmd, sshOptions, sshUserName, at, hostname, remoteCmd,
+          timeOutInterval/1000f);
+      LOG.info("Executing full command [" + cmd + "]");
+      return new String[] { "/usr/bin/env", "bash", "-c", cmd };
+    }
+
+    @Override
+    public void execute() throws IOException {
+      super.execute();
+    }
```

Comment:
Why is it necessary?

Reply:
You are right, it isn't. I'll remove it from both here and from RemoteShell.